design/namespaced-roles.md

```diff
@@ -0,0 +1,299 @@
+* **Status**: Draft #1
+* **JIRA**: TBD
+* **Github Discussion**: [Namespaced Roles](https://github.com/keycloak/keycloak/discussions/8516)
+
+## Motivation
+
+In Keycloak, roles are defined at the Realm and Client levels. These roles can then be assigned directly to a User or to a Group, making every user in that group inherit the role.
+
+This is done through role mapping tables in the database, limiting the way you can assign roles to those two use cases.
+
+We will need more flexibility in the future, for example, to assign a role to a user in the context of a specific group (instead of having everyone in the group inherit a role) or, as the tenancy model gets implemented, a way to both define and assign roles in the context of a tenant.
```

Comment:
Would also like the motivation to touch on things like:

* The need to create clients just to serve as a placeholder for roles, like we do for realm management roles
* The inability to share a common set of roles with a group of clients
* The inability to properly organize roles

Reply:
+1. I'll add those points.